src/saml2/entity.py

```diff
@@ -422,6 +422,10 @@ def sign(self, msg, mid=None, to_sign=None, sign_prepare=False,
             msg.signature = pre_signature_part(msg.id, self.sec.my_cert, 1,
                                                sign_alg=sign_alg,
                                                digest_alg=digest_alg)
+        
+        # sign adn digest algs
+        sign_alg = sign_alg or self.signing_algorithm
```

Comment:
shouldn't these two lines be before 
`if msg.signature is None:` ?
That's the only point of this function where `sign_alg` and `digest_alg` are used

Reply:
Oh-my-gosh ... What would i do without you? :smile: 
Fixed and ammended, so it won't introduce any additional commit